plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayerPlugin.java

```diff
@@ -489,6 +490,7 @@ public void onStop() {
     if (mapboxMap != null) {
       mapboxMap.removeOnCameraMoveListener(onCameraMoveListener);
     }
+    mapView.removeOnMapChangedListener(onMapChangedListener);
```

Comment:
Won't this one unregister the listener during style switch resulting in disabling the location layer? See https://github.com/mapbox/mapbox-plugins-android/blob/86a67081e4a188433cfd7f51740c9dfa2fafc43e/plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayerPlugin.java#L632-L641.

Reply:
that only happens with onMapReady callbacks not with end user provided map change listeners